src/client/pythonEnvironments/common/environmentIdentifier.ts

```diff
@@ -0,0 +1,160 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+import * as fsapi from 'fs-extra';
+import * as path from 'path';
+import { traceWarning } from '../../common/logger';
+import { createDeferred } from '../../common/utils/async';
+import { getEnvironmentVariable } from '../../common/utils/platform';
+import { EnvironmentType } from '../info';
+
+function pathExists(absPath: string): Promise<boolean> {
+    const deferred = createDeferred<boolean>();
+    fsapi.exists(absPath, (result) => {
+        deferred.resolve(result);
+    });
+    return deferred.promise;
+}
+
+/**
+ * Checks if the given interpreter path belongs to a conda environment. Using
+ * known folder layout, and presence of 'conda-meta' directory.
+ * @param {string} interpreterPath: Absolute path to any python interpreter.
+ *
+ * Remarks: This is what we will use to begin with. Another approach we can take
+ * here is to parse ~/.conda/environments.txt. This file will have list of conda
+ * environments. We can compare the interpreter path against the paths in that file.
+ * We don't want to rely on this file because it is an implementation detail of
+ * conda. If it turns out that the layout based identification is not sufficient
+ * that is the next alternative that is cheap.
+ *
+ * sample content of the ~/.conda/environments.txt:
+ * C:\envs\\myenv
+ * C:\ProgramData\Miniconda3
+ *
+ * Yet another approach is to use `conda env list --json` and compare the returned env
+ * list to see if the given interpreter path belongs to any of the returned environments.
+ * This approach is heavy, and involves running a binary. For now we decided not to
+ * take this approach, since it does not look like we need it.
+ *
+ * sample output from `conda env list --json`:
+ * conda env list --json
+ * {
+ *   "envs": [
+ *     "C:\\envs\\myenv",
+ *     "C:\\ProgramData\\Miniconda3"
+ *   ]
+ * }
+ */
+async function isCondaEnvironment(interpreterPath: string): Promise<boolean> {
+    const condaMetaDir = 'conda-meta';
+
+    // Check if the conda-meta directory is in the same directory as the interpreter.
+    // This layout is common in Windows.
+    // env
+    // |__ conda-meta  <--- check if this directory exists
+    // |__ python.exe  <--- interpreterPath
+    const conda_env_dir_1 = path.join(path.dirname(interpreterPath), condaMetaDir);
```

Comment:
Resurfacing @int19h's comment

```suggestion
    const condaEnvDir1 = path.join(path.dirname(interpreterPath), condaMetaDir);
```